### PR TITLE
Fix TPU circleci tests

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes TPU circleci tests

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda @carmocca @akihironitta @kaushikb11 @rohitgr7 @tchaton

### CODE REVIEW

**Comment by kaushikb11** [comment]
Need to fix this as well: https://cloud.google.com/blog/products/containers-kubernetes/kubectl-auth-changes-in-gke

**Review by Borda — APPROVED** [review]

File: src/pytorch_lightning/strategies/tpu_spawn.py [hunk 1]
```diff
@@ -77,6 +77,18 @@ def __init__(
...
+    def global_rank(self) -> int:
+        return self.tpu_global_core_rank
+
+    @property
+    def local_rank(self) -> int:
+        return self.tpu_local_core_rank
+
+    @property
+    def world_size(self) -> int:
+        return xm.xrt_world_size()
```
Comment (rohitgr7) [inline src/pytorch_lightning/strategies/tpu_spawn.py:90]: should it use xla cluster env?

**Review by rohitgr7 — APPROVED** [review]

File: src/pytorch_lightning/strategies/tpu_spawn.py (hunk 1 above)
Comment (kaushikb11) [reply src/pytorch_lightning/strategies/tpu_spawn.py:90]: With TPUs, you only get the right ranks with the XLA api inside the processes, whereas we are calling it before hand.
 So the local and global ranks are always set to 0. 
 Dealing withe XLA cluster env would be a follow up PR.

**Review by carmocca — APPROVED** [review]

File: src/pytorch_lightning/strategies/tpu_spawn.py (hunk 1 above)
Comment (kaushikb11) [reply src/pytorch_lightning/strategies/tpu_spawn.py:90]: that as well

**Review by awaelchli — COMMENTED** [review]
By overriding these properties, this reverts essentially the introduction of the XLAEnvironment #11321 entirely. In  #11163 I need to remove them again. 

Both of these PRs were passing the tests before the TPU CI was taken down (at the time I posted https://pytorch-lightning.slack.com/archives/G01001EF4M7/p1652577903723629 so there must be something we don't cover in the test suite.

**Review by akihironitta — APPROVED** [review]

**Pull request merged**